obal/data/modules/srpm.py

```diff
@@ -0,0 +1,89 @@
+"""
+Build SRPM
+"""
+
+import shutil
+import os
+import commands
+from tempfile import mkdtemp
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.chdir import chdir
+
+def run_command(command):
+    return commands.getstatusoutput(command)
+
+def copy_sources(spec_file, package_dir, sources_dir):
+    command = "spectool --list-files %s" % spec_file
+    _status, sources = run_command(command)
+
+    source_files = []
+
+    for source in sources.split("\n"):
+        if 'http' not in source:
+            shutil.copy("%s/%s" % (package_dir, source.split(' ')[1]), sources_dir)
+        else:
+            with chdir(package_dir):
+                run_command("git-annex lock")
+                _status, annexed_files = run_command("git-annex find --include='*'")
+                annexed_files = annexed_files.splitlines()
+                run_command("git-annex get")
+                run_command("git-annex unlock")
+
+                for annex in annexed_files:
+                    shutil.copy(annex, sources_dir)
+
+                run_command("git-annex lock --force")
+
+
+def main():
+    """
+    Build a package using tito
+    """
+    module = AnsibleModule(
+        argument_spec=dict(
+            package=dict(type='str', required=False),
+            output=dict(type='path', required=False),
+        )
+    )
+
+    package = module.params['package']
+    output = module.params['output']
+
+    spec_file = '%s/%s.spec' % (package, os.path.basename(package))
+
+    base_dir = mkdtemp()
+    sources_dir = os.path.join(base_dir, 'SOURCES')
+    build_dir = os.path.join(base_dir, 'BUILD')
+
+    os.mkdir(sources_dir)
+    os.mkdir(build_dir)
+
+    copy_sources(spec_file, package, sources_dir)
+    shutil.copy(spec_file, base_dir)
+
+    command = ['rpmbuild', '-bs']
+    command.append('--define "_topdir %s"' % base_dir)
+    command.append('--define "_sourcedir %s"' % sources_dir)
+    command.append('--define "_builddir %s"' % build_dir)
+    command.append('--define "_srcrpmdir %s"' % base_dir)
+    command.append('--define "_rpmdir %s"' % base_dir)
```

Comment:
I'd prefer something like:
```suggestion
    defines = {
        "_topdir": base_dir,
        "_sourcedir": sources_dir,
        "_builddir": build_dir,
        "_srcrpmdir": base_dir,
        "_rpmdir": base_dir,
    }
    for key, value in defines.items():
        command.append('--define')
        command.append("%s %s" % (key, value))
```

Then call the command as an array so you don't need to deal with quoting.

Reply:
I am calling it as an array already. Whats the problem with quoting? I'm open to the idea if there is a bug I am preventing. Otherwise I like the explicitness of the current method.

Reply:
Your current call is `status, result = commands.getstatusoutput(' '.join(command))` but I was suggesting to call it as `subprocess.check_call(command)` (where command is a list). In your version you don't quote the path so if there's a space in `base_dir`, it will fail. Granted, it's unlikely but I've taught myself that doing an array is a way of avoiding a whole class of problems (like using prepared statements instead of trying to quote SQL queries).

Reply:
Btw, if you want to spell it out and still use list, an alternative way of writing this is:
```python
command += ['--define', '_topdir %s' % base_dir]
command += ['--define', '_sourcedir %s' % sources_dir]
# ...
```

Reply:
Ohh right, I can re-test, but calling out to rpmbuild was having issues when I passed the command as an array.